das/done.go

```diff
@@ -0,0 +1,31 @@
+package das
+
+import (
+	"context"
+	"fmt"
+)
+
+type done struct {
+	name     string
+	finished chan struct{}
+}
+
+func newDone(name string) done {
+	return done{
+		name:     name,
+		finished: make(chan struct{}),
+	}
+}
+
+func (sm *done) Done() {
```

Comment:
Why is this exported?

Reply:
API of `done` could be bnefitial outside of package. There is another topic that @Wondertan mentioned previously in review: this generic primitive should live in some utility package outside of DASer, we jsut need to find good spot for it

Reply:
We generally should not export methods until they are actually useful outside the package.

Please no premature decisions like this, only leads to us asking "why is this thing exported again?" a couple months down the line

Reply:
There is no decision, and this is very minor. The public method on a private struct is the same as private, which is the case for the Done. This won't cause us any problems down the line.